core/citrus-base/src/main/java/org/citrusframework/actions/ReceiveMessageAction.java

```diff
@@ -289,31 +320,30 @@ protected void validateMessage(Message message, TestContext context) {
             if (!unknown.isEmpty()) {
                 unknown.forEach(validationContext -> logger.warn("Found validation context that has not been processed: {}", validationContext.getClass().getName()));
                 throw new ValidationException("Incomplete message validation of message with type %s - no validation has been performed on the following contexts: %s".formatted(messageType, unknown.stream().map(validationContext -> validationContext.getClass().getSimpleName()).collect(
-                    Collectors.joining(", "))));
+                        Collectors.joining(", "))));
             }
         }
     }
 
     private void assumeMessageType(Message message) {
-
         Object contentType = message.getHeaders().get("Content-Type");
         if (contentType instanceof String contentTypeString) {
             Object encoding = message.getHeaders().get("Content-Transfer-Encoding");
             MessageType messageType = mapToMessageType(
-                contentTypeString,
-                encoding != null ? encoding.toString() : null);
+                    contentTypeString,
+                    encoding != null ? encoding.toString() : null);
             if (messageType != null) {
                 setMessageType(messageType);
                 return;
             }
         }
 
         if (MessageType.isBinary(getMessageType()) ||
-            MessageType.FORM_URL_ENCODED.equalsIgnoreCase(getMessageType())) {
+                MessageType.FORM_URL_ENCODED.equalsIgnoreCase(getMessageType())) {
             return;
         }
 
-        var payload = message.getPayload(String.class);
+        var payload = isNull(message.getPayload()) && "null".equals(message.getPayload(String.class)) ? null : message.getPayload(String.class);
```

Comment:
Can't this be done better? It is smelly. Having a "null" string here seems to be the issue. Can't this be solved earlier? In my opinion message.getPayload(String.class) should never return "null" unless there is a payload which really contains "null"

Reply:
I think the problem was something along the line of this: a JSON message with `null`'ish payload resolves to `{}` when calling `Message#getPayload(String)` - but of course it does not have an `Object` representation, because it is an empty object. it is not "effectively nullish". thus a check on `Message#getPayload()` is not sufficient. it is additionally required to make sure the String representation is also "null".

I've refactored the code a bit, moved this into a method with a comment. that should make it more clear.

I don't think it can be catched earlier on, respectively this would go to deep and change too much of "accepted behavior".